src/test/requestUser.test.js

```diff
@@ -0,0 +1,263 @@
+import {
+  collection,
+  addDoc,
+  getDocs,
+  query,
+  where,
+  updateDoc,
+  arrayUnion,
+  arrayRemove,
+  doc,
+} from "firebase/firestore";
+
+import {
+  addUserRequest,
+  acceptRequest,
+  userPendingList,
+  userAcceptedList,
+  checkUserRequest,
+  checkUserPending,
+} from "@/firebase/requestUser";
+
+const tableName = "user_request";
+
+jest.mock("firebase/firestore", () => ({
+  getFirestore: jest.fn(),
+  collection: jest.fn(),
+  addDoc: jest.fn(),
+  query: jest.fn(),
+  getDocs: jest.fn(),
+  where: jest.fn(),
+  updateDoc: jest.fn(),
+  arrayUnion: jest.fn(),
+  arrayRemove: jest.fn(),
+}));
+
+describe("User Request logic", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  [
+    {
+      user_id: "1231",
+      pending_list: ["Walter", "Joe"],
+      accepted_list: ["White"],
+    },
+    {
+      uid: "1234",
+      pending_list: ["Joe"],
+      accepted_list: ["White"],
+    },
+  ];
```

Comment:
is this saved anywhere? is it used or can it be deleted?

Reply:
Yup this was just for reference i will remove this.